### PR TITLE
Fix types

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: contribs/gmf/src/filters/filterselectorComponent.js [hunk 1]
```diff
@@ -248,7 +248,7 @@ class Controller {
     /**
      * The data source ready to be filtered, after it has been selected and
      * prepared.
-     * @type {?import("gmf/datasource/OGC.js").default}
+     * @type {?import("gmf/datasource/OGC.js").OGCOptions}
```
Comment (adube) [inline contribs/gmf/src/filters/filterselectorComponent.js]: This is invalid.  The type of this property is not a hash of option. It's an instance of a class.

File: contribs/gmf/src/filters/filterselectorComponent.js [hunk 2]
```diff
@@ -541,8 +541,7 @@ class Controller {
   }
 
   /**
-   * @param {?import("gmf/datasource/OGC.js").default} dataSource Newly selected data source
-   *     object.
+   * @param {?import("gmf/datasource/OGC.js").OGCOptions} dataSource Newly selected data source object.
```
Comment (adube) [inline contribs/gmf/src/filters/filterselectorComponent.js]: idem

File: contribs/gmf/src/filters/filterselectorComponent.js [hunk 3]
```diff
@@ -563,9 +562,7 @@ class Controller {
       this.active = true;
     }
 
-    this.gmfDataSourcesHelper_.prepareFiltrableDataSource(
-      dataSource
-    ).then((dataSource) => {
+    this.gmfDataSourcesHelper_.prepareFiltrableDataSource(dataSource).then((dataSource) => {
```
Comment (adube) [inline contribs/gmf/src/filters/filterselectorComponent.js]: Why was this changed?  It was easier to read before.

File: contribs/gmf/src/filters/filterselectorComponent.js (hunk 1 above)
Comment (sbrunner) [reply contribs/gmf/src/filters/filterselectorComponent.js]: Right, fixed be removing one @extent :-), Thanks.

File: contribs/gmf/src/datasource/OGC.js [hunk 4]
```diff
@@ -4,7 +4,6 @@ import ngeoDatasourceOGC from 'ngeo/datasource/OGC.js';
 /**
  * The options required to create a `gmf.datasource.OGC`.
  * @typedef {Object} OGCOptions
- * @extends import('ngeo/datasource/OGC.js').OGCOptions
```
Comment (adube) [inline contribs/gmf/src/datasource/OGC.js]: Hum...  I don't understand how this is supposed to affect the type checking yet, but I won't prevent you from merge just because of this.  I'll figure it out later.

**Comment by adube** [comment]
Thanks for the fix.  Check what's wrong with Travis then merge.

**Comment by fredj** [comment]
Build restarted

File: contribs/gmf/src/datasource/OGC.js (hunk 4 above)
Comment (sbrunner) [reply contribs/gmf/src/datasource/OGC.js]: One good example of using extends is the following:
```
/**
 * @extends A<number>
 */
class B extends A {...}
```
To the value of the template.
In this case it's incoherent, then I don't know what it can mean...
